python/cudf/cudf/_lib/orc.pyx

```diff
@@ -123,8 +129,59 @@ cpdef read_orc(object filepaths_or_buffers,
         c_result = move(libcudf_read_orc(c_orc_reader_options))
 
     names = [name.decode() for name in c_result.metadata.column_names]
+    cdef map[string, string] user_data = c_result.metadata.user_data
```

Comment:
Maybe it would be good to place this code in a separate function, not sure.

Reply:
I agree.

Reply:
Apologies for the delay, moved this to a separate function.